tests/unit/aspect.ts

```diff
@@ -171,6 +171,41 @@ registerSuite({
 			}
 		},
 
+		'.on()': {
+			'advising function returns undefined, returns original result'() {
+				const aspectStub1 = sinon.stub();
+
+				aspect.on(obj, 'method', aspectStub1);
+
+				assert.strictEqual(obj.method(0), 1);
+				assert.isTrue(aspectStub1.calledAfter(methodSpy));
+			},
+
+			'advising function returns defined values, returns advising function result'() {
+				const aspectStub1 = sinon.stub().returns(2);
+
+				aspect.on(obj, 'method', aspectStub1);
+
+				assert.strictEqual(obj.method(0), 2);
```

Comment:
We're talking about `aspect.on`, so I'd potentially expect the return value to be meaningless.  It currently inherits `after` + `receiveArguments` logic from Dojo 1 so it can still influence the return value.

Do people often want to influence the return value when `receiveArguments` is true?  Usually it's more useful to have the returned value for processing when you actually want to change it (i.e. `receiveArguments` false / `aspect.after` in Dojo 2).

Another thing that is potentially awkward is that I would figure that right now, `after` and `on` are sharing the same queue.

I think these are things to consider but shouldn't block this PR.  But we may want to leave this behavior (of whether it modifies the return value) as undefined for now.